### ISSUE TITLE
Accessibility: BountyHeader search and button issues

### ISSUE THREAD

**Issue by jason-me** [stakwork/sphinx-tribes-frontend#1237]
# Accessibility: BountyHeader "Search", "Filter", and "Developers" issues

## Issue
When a user navigates to the BountyHeader on the Bounties Workspace page, there are accessibility issues with the "Search" `<input>` label relying on placeholder text alone (which also has insufficient contrast), and the component acts like an editable combobox with inline and list autocomplete but does not follow the established design pattern. 

Additionally, the clear search "X", as well as the "Filter" and "Developers" buttons are missing the appropriate `button` roles and associated keyboard behaviors. 

## Expected Behavior
- [WCAG 2.2 A 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG22/#name-role-value) requires that programmatically determined roles, states, and properties are communicated to screen reader users.
- [HTML5 Accessibility Chops: the placeholder attribute](https://www.tpgi.com/html5-accessibility-chops-the-placeholder-attribute/).
- [WAI - ARIA APG Editable Combobox With Both List and Inline Autocomplete Example Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/)
- [WAI - ARIA APG Button Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/button/)

### For the search `<input>` ###
- Screen reader users expect to hear the correct name of the input announced (e.g. "search"). This should be placed in a persistent visible `<label>` with a `for` attribute that references the `id` of the `<input>`. This visible label should have at least a 4.5:1 contrast ratio. **Note:** In this case, there is already a "Search" icon, but it is not persistent or programmatically associated with the input.

    - Additionally, the associated autcomplete should follow the established design pattern for an [Editable Combobox With Both List and Inline Autocomplete](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/) or a similar accessible design pattern.

    - Keyboard users expect to encounter the [expected keyboard interaction for comboboxes](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboardinteraction)

### For `button`s ###
- Screen reader users expect to hear the `button` role announced along with its accessible name (e.g. "Clear search", "Filter(2)", or "Developers (480)"), not the filename of the icon.
- Keyboard users expect to encounter the [expected keyboard interaction for buttons](https://www.w3.org/WAI/ARIA/apg/patterns/button/#keyboardinteraction) 

### For the `<input>` ###
- [ ] A persistent visible label with a contrast ratio of at least 4.5:1 is present or 3:1 for a visible search icon
- [ ] The `<label for` attribute is programmatically associated with the persistent visible label or icon with text. 
- [ ] The the associated autcomplete should follow the established design pattern for an [Editable Combobox With Both List and Inline Autocomplete](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/) or a similar accessible design pattern.
- [ ] Test with a screen reader JAWS, VoiceOver, NVDA, Orca, Talkback, etc. - ensure the correct accessible name and role(s) are announced.
- [ ] Test with a keyboard to ensure the pattern follows the [expected keyboard interaction for comboboxes](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboardinteraction)
-  [ ] Verify in Chrome and Firefox with and without a mobile viewport enabled to ensure visual look and feel is unchanged.

### For `button`s ###
- [ ] The button is a native `button` element or has `role="button"`.
- [ ] The button has an accessible label. By default, the accessible name is computed from any text content inside the button element. However, it can also be provided with `aria-labelledby` or `aria-label`.
- [ ] If a description of the button's function is present, the button element has `aria-describedby` set to the `id` of the element containing the description.
- [ ] When the action associated with a button is unavailable, the button has `aria-disabled` set to `true`.
- [ ] Test with a screen reader JAWS, VoiceOver, NVDA, Orca, Talkback, etc. - ensure the correct name and `button` role are announced.
- [ ] Test with a keyboard to ensure the pattern follows the [expected keyboard interaction for buttons](https://www.w3.org/WAI/ARIA/apg/patterns/button/#keyboardinteraction) 
-  [ ] Verify in Chrome and Firefox with and without a mobile viewport enabled to ensure visual look and feel is unchanged.

_Linked to tracking issue: #1233_
_[Bounty#3990](https://community.sphinx.chat/bounty/3990)_

**Comment by jason-me** [comment]
Closed by #1382
